proxy/src/bin/local_proxy.rs

```diff
@@ -139,12 +149,28 @@ async fn main() -> anyhow::Result<()> {
         16,
     ));
 
-    refresh_config(args.config_path.clone()).await;
+    // write the process ID to a file so that compute-ctl can find our process later
+    // in order to trigger the appropriate SIGHUP on config change.
+    let pid = std::process::id();
+    info!("process running in PID {pid}");
+    std::fs::write(args.pid_path, format!("{pid}\n")).context("writing PID to file")?;
 
     let mut maintenance_tasks = JoinSet::new();
+
+    let refresh_config_notify = Arc::new(Notify::new());
+    let refresh_config_notify2 = Arc::clone(&refresh_config_notify);
```

Comment:
nit: you could introduce a scoped block that contains the clone and the handle_signals call instead of having a numbered clone.